### PR TITLE
Load magnet url instead of searching

### PR DESCRIPTION
**Pull request by simonhong**
Fix https://github.com/brave/brave-browser/issues/1435

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
`yarn test brave_browser_tests --filter=BraveContentBrowserClientTest.TypedMagnetURL`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/autocomplete/brave_autocomplete_scheme_classifier.cc [hunk 1]
```diff
@@ -25,7 +25,8 @@ BraveAutocompleteSchemeClassifier::GetInputTypeForScheme(
     return metrics::OmniboxInputType::INVALID;
   }
   if (base::IsStringASCII(scheme) &&
-       base::LowerCaseEqualsASCII(scheme, kBraveUIScheme)) {
+      (base::LowerCaseEqualsASCII(scheme, kBraveUIScheme) ||
+       base::LowerCaseEqualsASCII(scheme, kMagnetScheme))) {
```
Comment (yrliou) [inline browser/autocomplete/brave_autocomplete_scheme_classifier.cc]: Could you check whether WebTorrent extension is enabled here, probably through ExtensionRegistry?
We could just use chromium's result when WebTorrent is disabled, and only change it when it is enabled, wdyt?

Comment (simonhong) [reply browser/autocomplete/brave_autocomplete_scheme_classifier.cc]: Do you mean `IsWebtorrentEnabled(profile) && base::LowerCaseEqualsASCII(scheme, kMagnetScheme)`?

Comment (yrliou) [reply browser/autocomplete/brave_autocomplete_scheme_classifier.cc]: right

Comment (simonhong) [reply browser/autocomplete/brave_autocomplete_scheme_classifier.cc]: I'm trying to include `content_browser_client_helper.h` in this cc file but I got some compile error. Did you intend this header is only included in `brave_content_browser_client.cc`?
I'm curious why you implemented all methods in header.

Comment (simonhong) [reply browser/autocomplete/brave_autocomplete_scheme_classifier.cc]: Done. `IsWebtorrentEnabled()` changed to api.

**Comment by yrliou** [comment]
master: d2c57378b75c4697244dc2db7ccace7cac692166
0.58.x: df2caaee30f3dc4c54655f554cd6a7e3aafbd708

**Pull request merged**